Bottom: a26452a4d2b2149f6734879a73eb7a2f66dc5492
Top:    e5af899438bf61af290d32768e6d01c95e12a4fc
Author: student <student@ReliableEmbeddedSystems.com>
Date:   2019-09-10 15:16:55 +0200

use systemd as cgroup manager for docker - While it's possible to configure docker and kubelet to use cgroupfs this means that there will then be two different cgroup managers, so we use systemd only

Signed-off-by: student <student@ReliableEmbeddedSystems.com>


---

diff --git a/recipes-containers/docker/docker-ce_git.bb b/recipes-containers/docker/docker-ce_git.bb
index 042d3a4..2dac4a8 100644
--- a/recipes-containers/docker/docker-ce_git.bb
+++ b/recipes-containers/docker/docker-ce_git.bb
@@ -95,6 +95,18 @@ do_compile() {
 	VERSION="${DOCKER_VERSION}" DOCKER_GITCOMMIT="${SRCREV_docker}" make dynbinary
 }
 
+# use /data/docker as a data dir for docker, since data is a btrfs partition:
+# --data-root="/data/docker"
+# While it's possible to configure docker and kubelet to use cgroupfs this means
+# that there will then be two different cgroup managers, so we use systemd only:
+# --exec-opt native.cgroupdriver=systemd
+do_install_append() {
+        if ${@bb.utils.contains('DISTRO_FEATURES','systemd','true','false',d)}; then
+            #sed -i 's/ExecStart=\/usr\/bin\/dockerd -H fd:\/\//ExecStart=\/usr\/bin\/dockerd -H fd:\/\/ --data-root="\/data\/docker" --exec-opt native.cgroupdriver=systemd/' ${D}/${systemd_unitdir}/system/docker.service
+            sed -i 's/ExecStart=\/usr\/bin\/dockerd -H fd:\/\//ExecStart=\/usr\/bin\/dockerd -H fd:\/\/ --data-root="\/data\/docker"/' ${D}/${systemd_unitdir}/system/docker.service
+        fi
+}
+
 do_install() {
 	mkdir -p ${D}/${bindir}
 	cp ${S}/src/import/components/cli/build/docker ${D}/${bindir}/docker
diff --git a/recipes-containers/docker/files/daemon.json b/recipes-containers/docker/files/daemon.json
new file mode 100644
index 0000000..5d18abc
--- /dev/null
+++ b/recipes-containers/docker/files/daemon.json
@@ -0,0 +1,8 @@
+{
+  "exec-opts": ["native.cgroupdriver=systemd"],
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "100m"
+  },
+  "storage-driver": "overlay2"
+}
